### ISSUE TITLE
Some unittests fail on qemux86-64 (yocto)

### ISSUE THREAD

**Issue by anyj0527** [nnstreamer/nntrainer#658]
I'm adding yocto recipe for nntrainer: https://github.com/nnstreamer/meta-neural-network/pull/56

There are 2 fail cases in two test suites: `unittest_nntrainer_layers` and `unittest_tizen_capi` in qemux86-64.
Logs:
- unittest_nntrainer_layers - 1 fail (nntrainer_Conv2DLayer.backwarding_03_p)
```
[ RUN      ] nntrainer_Conv2DLayer.backwarding_03_p
../git/test/unittest/unittest_nntrainer_layers.cpp:91: Failure
The difference between out_ptr[i] and golden_ptr[i] is 4, which exceeds local_tolerance, where
out_ptr[i] evaluates to 60843296,
golden_ptr[i] evaluates to 60843292, and
local_tolerance evaluates to 9.9999997473787516e-05.
../git/test/unittest/unittest_nntrainer_layers.cpp:91: Failure
The difference between out_ptr[i] and golden_ptr[i] is 4, which exceeds local_tolerance, where
out_ptr[i] evaluates to 64254944,
golden_ptr[i] evaluates to 64254948, and
local_tolerance evaluates to 9.9999997473787516e-05.

......

../git/test/unittest/unittest_nntrainer_layers.cpp:91: Failure
The difference between out_ptr[i] and golden_ptr[i] is 4, which exceeds local_tolerance, where
out_ptr[i] evaluates to 51036928,
golden_ptr[i] evaluates to 51036932, and
local_tolerance evaluates to 9.9999997473787516e-05.
../git/test/unittest/unittest_nntrainer_layers.cpp:91: Failure
The difference between out_ptr[i] and golden_ptr[i] is 8, which exceeds local_tolerance, where
out_ptr[i] evaluates to 66626664,
golden_ptr[i] evaluates to 66626656, and
local_tolerance evaluates to 9.9999997473787516e-05.
[  FAILED  ] nntrainer_Conv2DLayer.backwarding_03_p (155 ms)
```

- unittest_tizen_capi - 1 fail (nntrainer_capi_nnmodel.train_01_p)
```
[ RUN      ] nntrainer_capi_nnmodel.train_01_p
#1/1 - Training Loss: 4.01373 >> [ Accuracy: 10.4167% - Validation Loss : 3.55133 ] 
../git/test/tizen_capi/unittest_tizen_capi.cpp:51: Failure
Expected equality of these values:
  std::strtof(summary2, nullptr)
    Which is: 3.5513301
  valid_loss
    Which is: 3.5513401
[  FAILED  ] nntrainer_capi_nnmodel.train_01_p (8027 ms)
```

This could be the bug of qemu or OpenBLAS ?

**Comment by kparichay** [comment]
This is from the calculation precision difference with different versions of blas/different hardware etc.
If you some time to spare, for the unittest `nntrainer_capi_nnmodel.train_01_p`, please change the epochs to 100 (add statement `replaceString("Epochs = 1", "Epochs = 100", config_file, config_str);`).
On ubuntu machine local build, the training and validation loss are 2.0330501 & 2.31356 respectively. Same for tizen on Qemu.
If the different result in yocto build is because of the difference in precision of library, then the order of difference should remain the same, and we can lower the error tolerance for these unittests. If the order difference has increased, then IMO, you have exposed a bug in nntrainer.

**Comment by anyj0527** [comment]
```
#100/100 - Training Loss: 2.03305 >> [ Accuracy: 10.4167% - Validation Loss : 2.31356 ]
../git/test/tizen_capi/unittest_tizen_capi.cpp:43: Failure
Expected equality of these values:
  std::strtof(summary1, nullptr)
    Which is: **2.0330501**
  train_loss
    Which is: 4.01373
../git/test/tizen_capi/unittest_tizen_capi.cpp:51: Failure
Expected equality of these values:
  std::strtof(summary2, nullptr)
    Which is: **2.31356**
  valid_loss
    Which is: 3.5513401
[  FAILED  ] nntrainer_capi_nnmodel.train_01_p (684491 ms)
```

Got `2.0330501` & `2.31356` respectively, which are same as values in other environments.
Lowering the error tolerance would be enough. Thanks!
